Sources/FoundationEssentials/Platform.swift

```diff
@@ -172,31 +171,46 @@ extension Platform {
     
     static func name(forUID uid: uid_t) -> String? {
         withUserGroupBuffer(uid, passwd(), sizeProperty: Int32(_SC_GETPW_R_SIZE_MAX), operation: getpwuid_r) {
-            String(cString: $0.pw_name)
+            // Android's pw_name `char *`` is nullable, so always coerce to a nullable pointer
```

Comment:
Just to confirm: is this property nullable on `Android` because it has different semantics than the Darwin/Glibc definition (i.e. it is valid to be null whereas it's not valid on Darwin/Linux)? Or is this a potential issue with the Android module/swift overlay that we could address there to make it non-optional like Darwin/Linux?

If the difference is intentional then this seems like the best way to account for it, just want to make sure we shouldn't try to make this change to the Android module before we add complexity to this code that we don't have much unit testing for

Reply:
Good question! I looked over the sources for Android's libc and it looks like the situation is mixed:
- `pw_name` doesn't seem to be nullable , so I think it should be _Nonnull instead.
- `pw_gecos` is typically null, so it's correct for it to be _Nullable.
- `gr_name` doesn't seem to be nullable, so I think it should be _Nonnull instead.
- `pw_dir` doesn't seem to be nullable, so I think it should be _Nonnul linstead.

I'll update the patch to only account for nullability in `pw_gecos` in that case, and will try to re-annotate the other fields using API notes instead.

Reply:
Also `pw_gecos` isn't available on 32 bit android, as it's a macro in headers, so I'll handle that case too.

Reply:
the trouble with fields like this is "_when_ are making your claim?". even if one particular function guarantees the field won't be null, _other_ code can pass you an instance of this struct with a null field. there's no prohibition against user code creating/mutating instances of any libc type (nor could there be, in C).

tbh, i don't think there's any particularly satisfactory answer to this question. afaik no other libc has tried adding annotations, so we can't even compare notes to be "at least consistently wrong"...

i'm listening, but this feels "WAI" to me, given that insufficient expressivity of the existing annotations.

(and if you think this is bad, check out the places where we've shrugged and said `__BIONIC_COMPLICATED_NULLNESS` where the answer is "it depends on the values of other arguments" :-( )